include/aws/common/common.h

```diff
@@ -78,12 +78,26 @@ AWS_COMMON_API void aws_load_error_strings(void);
 
 #define AWS_CACHE_ALIGN AWS_ALIGN(AWS_CACHE_LINE)
 
+#define AWS_OP_SUCCESS 0
+#define AWS_OP_ERR -1
+
 typedef enum aws_common_error {
     AWS_ERROR_SUCCESS = 0,
     AWS_ERROR_OOM,
     AWS_ERROR_INVALID_BUFFER_SIZE,
     AWS_ERROR_INVALID_HEX_STR,
     AWS_ERROR_INVALID_INDEX,
+    AWS_ERROR_THREAD_INVALID_SETTINGS,
+    AWS_ERROR_THREAD_INSUFFICIENT_RESOURCE,
+    AWS_ERROR_THREAD_NO_PERMISSIONS,
+    AWS_ERROR_THREAD_NOT_JOINABLE,
+    AWS_ERROR_THREAD_NO_SUCH_THREAD_ID,
+    AWS_ERROR_THREAD_DEADLOCK_DETECTED,
+    AWS_ERROR_MUTEX_NOT_INIT,
+    AWS_ERROR_MUTEX_TIMEOUT,
+    AWS_ERROR_MUTEX_CALLER_NOT_OWNER,
+    AWS_ERROR_MUTEX_FAILED,
+    AWS_ERROR_CLOCK_FAILURE,
```

Comment:
I'd suggest having an AWS_ERROR_UNKNOWN instead of an AWS_[subsystem]_FAILURE generic error code.

Reply:
CLOCK or MUTEX failed is more useful when you queue up the errors or send out metrics than a generic unknown though.